File: Dedicate-labors/CalculatorUnitTest/unittest1.cpp
#include "stdafx.h"
#include "CppUnitTest.h"
#include"../Dedicate-labors/Calculator.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;

namespace CalculatorUnitTest
{		
	TEST_CLASS(UnitTest1)
	{
	public:
		
		TEST_METHOD(TestMethod1)
		{
			// TODO: 在此输入测试代码
			Calculator* calc = new Calculator();
			std::string ret = calc->Solve("11+22");
			Assert::AreEqual(ret, (std::string)"11+22=33");
		}

	};
}

File: 2stLv/Calculator/CalculatorUnitTestProject/UnitTest1.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace CalculatorUnitTestProject
{
    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public void TestMethod1()
        {
            Assert.AreEqual(1, Calculator.Program.Judge(50));
            //Assert.AreEqual(0, Calculator.Program.Judge(2.5));
            //Assert.AreEqual(0, Calculator.Program.Judge(-1));
        }
    }
}
